### DISCUSSION TITLE
About powerDistribution

### DISCUSSION THREAD

**Discussion by seuflyer** [bitcraze/discussions#1406] [Q&A]
Will there be significant differences in controller parameters between the two schemes of controllModeLegacy and controllModeForceTorque when using the same controller? Is the second option more reasonable?
![VE8~5T7Z72)U`RJ(IHEG5RV](https://github.com/bitcraze/discussions/assets/171215615/598ad539-34c4-4500-a592-f69fb3cf0397)

**Comment by knmcguire** [comment]
Hi!

So these control modes are depended on the output of the controller. If the controller outputs Force or ForceTorque, than those control modes are selected for that controller. Currently there are only 2 that do that, namely Lee's controller and Brescianini. All the others (PID, Mellinger, INDI) don't output torque or force but some unitless value that has something to do with the attitude and thrust commands. These are values that are very highly depended on the hardware so that makes it a bit difficult to use these controllers for any other platform than the Crazyflie or for separation of hardware and software for software in the loop.

Hope that makes it clear? It's not about which is the best but what your controller outputs as control command

**Reply by seuflyer** [reply]
Thank you very much for your answer.
But my question is that most control methods consider attitude control as a second-order system control problem. If PID is used for control, the physical meaning of u should be torque. So from the perspective of the model, it seems that the controller's output torque is more reasonable. But if the controller outputs an unitless signal, is it difficult to find a suitable explanation from the model?

**Reply by knmcguire** [reply]
Technically a PID controller's output can be anything the implementer want it to be. It's a generic controller that is used for all kind of processes outside of robotics as well.

The current implementation of the Crazyflie's PID controller output is not torque, but more closer to the PWM output of the motors before motor-mixing. The 'more correct way' would be to have the output to be torque, translate it to pwm and than send it to the motors after motormixing, but that would require retuning of the PID controller as the values that come out of it now are too high.

**Reply by seuflyer** [reply]
If I use the controller to directly output torque, where can I find the reference PID value? Relying on experience to adjust parameters is really too troublesome. Thank you.

**Reply by gemenerik** [reply]
The PID controller does not directly output torque. Therefore, there is no reference PID value related to torque. If I’ve misunderstood your question, please clarify what specifically you’re asking for.
